src/Data/PQueue/Internals/Foldable.hs

```diff
@@ -0,0 +1,22 @@
+-- | Writing 'Foldable' instances for nonuniform types in the natural manner
```

Comment:
```suggestion
-- | Writing 'Foldable' instances for non-uniform types (e.g. 'BinomForest') in the natural manner
```
I assume by non-uniform you mean polymorphic recursive?

Reply:
I meant non-regular, sorry. Using non-regular (AKA nested) types tends to require polymorphic recursive functions. I'll fix the terminology.